arches/app/media/js/viewmodels/reference-select.js

```diff
@@ -0,0 +1,158 @@
+define([
+    'jquery',
+    'knockout',
+    'knockout-mapping',
+    'arches',
+    'viewmodels/widget',
+], function($, ko, koMapping, arches, WidgetViewModel) {
+    var NAME_LOOKUP = {};
+    var ReferenceSelectViewModel = function(params) {
+        var self = this;
+
+        params.configKeys = ['placeholder'];
+        this.multiple = !!ko.unwrap(params.node.config.multiValue);
+        this.displayName = ko.observable('');
+        this.selectionValue = ko.observable([]); // formatted version of this.value that select2 can use
+        this.activeLanguage = arches.activeLanguage;
+
+        WidgetViewModel.apply(this, [params]);
+
+        this.getPrefLabel = function(labels){
+            return koMapping.toJS(labels)?.find(label => label.language===arches.activeLanguage && label.valuetype === 'prefLabel').value;
+        }; 
+
+        this.flattenItems = function(items, flatList, uniqueItems){
```

Comment:
Trying to remember what we said about this. Were you planning to just use the flat representation from the endpoint instead of flattening here?

Reply:
Good question - I just left this as it was, but I could switch to the flat list as long as we are confident that the list is ordered with children always following their parents.

Reply:
Yeah, we should be alright. I added a [comment](https://github.com/archesproject/arches/commit/7b655e0ab9f7319af2b7a8de9a9022e4d38ed393) next to the validation on sortorder -- sortorder concerns the flattened list as a whole. And the endpoint sorts on it.